app/javascript/components/Articles/ArticlesDetails.tsx

```diff
@@ -5,28 +5,38 @@ export interface ArticlesDetailsProps {
     content:string,
     firstName:string,
     lastName?:string,
-    created_at:string,
+    createdAt:string,
     link:string,
-    linkText:string
+    linkText:string,
+    editLink?:string,
+    currentUser?:number
 }
 
 const ArticlesDetails: React.FC<ArticlesDetailsProps> = (props) => {
+
+    let loggedInDisplay = null
+
+    if (props.currentUser > 0) {
+        loggedInDisplay = (<a href={props.editLink}>Edit Article</a> )
+    } 
+
     return (
         
-        <div className="outer-article-container">
+        <div className="outer-article-container" data-testid="article-details">
```

Comment:
Rather than using a data-testid, it would be possible to get this using `container.querySelector` again, as it has a class name you could grab by. Again, not essential, it just stops testids being necessary in code.

Reply:
I think I'll leave this in the project as an example of testids, but I see the power of querySelectors now.